### PR TITLE
Doc improvements for integrations and deployments

### PR DESCRIPTION
**Pull request by timothyhinrichs**
Debated whether putting this content into the integrations and deployment pages or elsewhere.  The integrations comparison feels pretty natural.  The deployment docs feels less so, but not bad where it is; suggestions welcome.

### CODE REVIEW

File: docs/content/integration.md [hunk 1]
```diff
@@ -9,14 +9,21 @@ OPA exposes domain-agnostic APIs that your service can call to manage and
 enforce policies. Read this page if you want to integrate an application,
 service, or tool with OPA.
 
+When integrating with OPA there are two interfaces to consider:
+* **evaluation**: the interface for asking for policy decisions.  This is primarily
+* **management**: the interface for deploying policies, understanding status, uploading logs, and so on
```
Comment (ashutosh-narkar) [inline docs/content/integration.md:14]: Nit: the bullet points did not render correctly.

File: docs/content/deployments.md [hunk 2]
```diff
@@ -315,4 +315,55 @@ variables.
...
+The relevance of the discussion above depends on how you have chosen to deploy policies into OPA.
+
+If you deploy policies to OPA on disk (e.g. volume mounting into the OPA container on Kubernetes), then OPA will only start answering policy queries once all the policies are successfully loaded.  In this case, it is impossible for OPA to answer policy queries before it has loaded policy, so the discussion above is a non-issue.
+
+On the other hand, if you use the [Bundle service](management.md#Bundles) OPA will start up without any policies and immediately start downloading a bundle.  But even before the bundle has successfully downloaded, OPA will answer policy queries if asked (which is in every case except the bootstrap case the right thing to do).  For this reason, OPA provides a `/health` [API](rest-api.md#health-api) that verifies that the server is operational and optionally that a bundle has been successfully activated.  As long as no policy queries are routed to OPA until the `/health` API verifies that OPA is operational.  The recommendation is to ensure the `/health` API indicates that OPA is operational before routing policy queries to it.
```
Comment (ashutosh-narkar) [inline docs/content/deployments.md:353]: `management.md#Bundles` > `../management.md#Bundles`

`rest-api.md#health-api` > `../rest-api.md#health-api`

File: docs/content/deployments.md [hunk 3]
```diff
@@ -315,4 +315,55 @@ variables.
...
+
+Finally, you might choose to push policies into OPA via its [REST API](restapi.md).  In this case, there is no way for OPA to know whether it has a complete policy set, and so the decision as to when to route policy queries to OPA must be handled by whatever software is pushing policies into OPA.
```
Comment (ashutosh-narkar) [inline docs/content/deployments.md:355]: `restapi.md` > `../rest-api`

File: docs/content/integration.md [hunk 4]
```diff
@@ -9,14 +9,21 @@ OPA exposes domain-agnostic APIs that your service can call to manage and
 enforce policies. Read this page if you want to integrate an application,
 service, or tool with OPA.
 
+When integrating with OPA there are two interfaces to consider:
+* **evaluation**: the interface for asking for policy decisions.  This is primarily
```
Comment (tsandall) [inline docs/content/integration.md:13]: This sentence is cut off.

File: docs/content/deployments.md [hunk 5]
```diff
@@ -315,4 +315,55 @@ variables.
...
+
+### Making Decisions before OPA is Ready
+
+The mechanisms discussed above ensure that OPA is not asked to answer policy queries before it is ready to do so.  But from the perspective of the software needing decisions, until OPA is operational, the software must make a decision on its own.  Typically there are two choices:
+* fail-open: if OPA does not provide a decision, then treat the decision as allowed.
+* fail-closed: if OPA does not provide a decision, then treat the decision as denied.
```
[the remark this answers is not in the document]
Comment (timothyhinrichs) [reply docs/content/deployments.md:361]: Possibly fixed.  Having trouble again running the site locally, and editor's markdown renders it correctly.  Will double-check once netlify preview is up.

File: docs/content/integration.md (hunk 1 above)
Comment (timothyhinrichs) [reply docs/content/integration.md:14]: Fixed.

File: docs/content/deployments.md (hunk 2 above)
Comment (timothyhinrichs) [reply docs/content/deployments.md:353]: Done

File: docs/content/integration.md (hunk 4 above)
Comment (timothyhinrichs) [reply docs/content/integration.md:13]: Fixed

File: docs/content/deployments.md (hunk 2 above)
Comment (tsandall) [reply docs/content/deployments.md]: Need to remove the `.md` extensions I think.

File: docs/content/deployments.md (hunk 3 above)
Comment (tsandall) [reply docs/content/deployments.md]: Same as above. Remove the .md extensions.

**Comment by timothyhinrichs** [comment]
Added a writeup to the FAQ about different policy styles, as requested.

**Comment by timothyhinrichs** [comment]
Also checked the links locally this time.

File: docs/content/integration.md [hunk 6]
```diff
@@ -9,14 +9,30 @@ OPA exposes domain-agnostic APIs that your service can call to manage and
...
+* **Management**: OPA's interface for deploying policies, understanding status, uploading logs, and so on.  This integration is typically the same across all OPA instances, regardless what software the evaluation interface is integrated with.  Distributing policy, retrieving status, and storing logs in the same way across all OPAs provides a unified management plane for policy across many different software systems.
+
+This page focuses predominantly on different ways to integrate with OPA's policy evaluation interface and how they compare.  For more information about the management interface:
+
+- See the [Bundle API](../management/#bundles) for distributing policy and data to OPA.
+- See the [Status API](../management/#status) for collecting status reports on bundle activation and agent health.
+- See the [Decision Log API](../management/#decision-logs) for collecting a log of policy decisions made by agents.
+- See the [Health API](../rest-api#health-api) for checking agent deployment readiness and health.
+- See the [Prometheus API endpoint](../management/#prometheus) to obtain insight into performance and errors.
```
Comment (ashutosh-narkar) [inline docs/content/integration.md:23]: Nit: `../monitoring/#prometheus`

**Pull request merged**